php/classes/controllers/class-players-controller.php

```diff
@@ -0,0 +1,181 @@
+		 * @todo ideally only when the player is loaded...
+		 */
+		add_action( 'wp_enqueue_scripts', array( $this, 'load_player_assets' ) );
+	}
+
+	public function load_player_assets() {
+		wp_register_style( 'html5-player-v2', $this->assets_url . 'css/html5-player-v2.css', array(), $this->version );
+		wp_enqueue_style( 'html5-player-v2' );
+		wp_register_script( 'html5-player-v2', $this->assets_url . 'js/html5-player-v2.js', array( 'jquery' ), $this->version, true );
+		wp_enqueue_script( 'html5-player-v2' );
+	}
+
+
+	public function register_shortcodes() {
+		add_shortcode('elementor_html_player', array($this, 'elementor_html_player'));
+	}
+
+	public function elementor_html_player($attributes) {
+		$templateData = $this->html_player($attributes['id']);
+
+		return $this->renderer->render($templateData, 'players/html-player');
+	}
+
+	/**
+	 * Return feed url.
+	 *
+	 * @return string
+	 */
+	public function get_feed_url() {
+		// Get feed slug
+		$feed_slug = apply_filters( 'ssp_feed_slug', $this->token );
+
+		if ( get_option( 'permalink_structure' ) ) {
+			$feed_url = $this->home_url . 'feed/' . $feed_slug;
+		} else {
+			$feed_url = $this->home_url . '?feed=' . $feed_slug;
+		}
+
+		$custom_feed_url = get_option( 'ss_podcasting_feed_url' );
+		if ( $custom_feed_url ) {
+			$feed_url = $custom_feed_url;
+		}
+
+		$feed_url = apply_filters( 'ssp_feed_url', $feed_url );
+
+		return $feed_url;
+	}
+
+	/**
+	 * Return html player for a given podcast (episode) id.
+	 *
+	 * @param int $id
+	 *
+	 * @return string
+	 */
+	public function html_player( $id ) {
+		$episode         = get_post( $id );
+		$episodeDuration = get_post_meta( $id, 'duration', true );
+		$audioFile       = get_post_meta( $id, 'audio_file', true );
+		$albumArt        = $this->episode_controller->get_album_art( $id );
+		$podcastTitle    = get_option( 'ss_podcasting_data_title' );
+
+		$subscribeLinks = $this->subscribe_links( $id );
+
+		$feedUrl = $this->get_feed_url();
+		// set any other info
+		$templateData = array(
+			'episode'      => $episode,
+			'duration'     => $episodeDuration,
+			'audioFile'    => $audioFile,
+			'albumArt'     => $albumArt,
+			'podcastTitle' => $podcastTitle,
+			'feedUrl'      => $feedUrl,
+			'itunes'       => $subscribeLinks['itunes'],
+			'stitcher'     => $subscribeLinks['stitcher'],
+			'spotify'      => $subscribeLinks['spotify'],
+			'googlePlay'   => $subscribeLinks['googlePlay']
+		);
+
+		$templateData = apply_filters( 'ssp_html_player_data', $templateData );
+
+		return $templateData;
+
+	}
+
+	/**
+	 * Return media player for a given podcast (episode) id.
+	 *
+	 * @param int $id
+	 *
+	 * @return string
+	 */
+	public function media_player( $id ) {
+		// get src file
+		$srcFile = get_post_meta( $id, 'audio_file', true );
+		$params  = array(
+			'src'     => $srcFile,
+			'preload' => 'none'
+		);
+
+		$mediaPlayer = wp_audio_shortcode( $params );
+
+		return $mediaPlayer;
+	}
+
+	public function subscribe_links( $id ) {
```

Comment:
I see you're using the subscribe_links method for the player. However, we also need to be able to render the subscribe links as a separate html template for an elementor widget. https://www.notion.so/castos/Third-Party-Plugin-theme-integrations-2dfc5d4cff574c39a210e645858b6a64#b7a8623ef4f348c79eaf47d68049a423

Reply:
@daniloradovic feel free to ping me on this one, if you're not sure what I mean.